product_filters/models/product.py

```diff
@@ -74,10 +75,11 @@ def _get_async_ranges(self, category):
         ranges_obj = self.env['product.price.ranges'].search([])
         count_dict = {}
         prod_ids = []
-        if category:
-            prod_ids = prod_obj.search(
-                [('public_categ_ids', 'child_of', int(category)),
-                 ('website_published', '=', True)])
+        if not category:
+            return []
+        prod_ids = prod_obj.search(
+            [('public_categ_ids', 'child_of', int(category)),
+             ('website_published', '=', True)])
         if prod_ids:
```

Comment:
Better use a final `return []` and remove the `if` sentence.
If the number of items in the `for prod in prod_ids:` is length zero then you can remove the `if not prod_ids`